### PR TITLE
refactor: Major refactoring - Rename 'agent' to 'tool' and improve architecture

### PR DESCRIPTION
**Pull request by lambdalisue**
## Summary

This PR contains a comprehensive refactoring of nvim-aibo to better reflect its versatility as a plugin for any interactive CLI tool, not just AI agents.

### Key Changes:
- **Terminology Update**: Renamed all references from "agent" to "tool" throughout the codebase
- **Module Reorganization**: Split and improved core modules for better separation of concerns
- **Test Suite Enhancement**: Increased test coverage from 108 to 135 tests with better isolation

### 1. Terminology Update: Agent → Tool
- **Rationale**: The plugin works with any interactive CLI tool (not just AI agents)
- **Configuration change**: `agents = {}` → `tools = {}`
- **File renaming**: `aibo-agent-*.lua` → `aibo-tool-*.lua`
- **API updates**: `get_agent_config()` → `get_tool_config()`
- **Documentation**: Updated all references in README, help files, and comments

#### Split Modules for Better Separation
- `console.lua` → Split into:
  - `console_window.lua`: Window and terminal management
  - `prompt_window.lua`: Prompt-specific functionality
- Removed redundant modules:
  - `controller.lua`: Merged into window modules
  - `send.lua`: Integrated into prompt_window

#### New Modules Added
- `timing.lua`: Debounce and throttle utilities
- `integration/init.lua`: Centralized integration management

#### Test Organization
- Moved `tests/unit/` → `tests/internal/`
- Renamed `test_runner.lua` → `runner.lua`
- Renamed `helpers.lua` → `mock.lua`

#### Coverage Improvements
- **Total tests**: 108 → 135
- **New test areas**:
  - Edge cases for empty buffers
  - Multiple console selection
  - Command option combinations
  - Integration health checks
  - Timing utilities

#### Test Stability
- Fixed vim.ui.select blocking in headless mode
- Improved timing tests with condition-based waiting
- Better test isolation using tabs

#### Console Window Module
- Added comprehensive error handling
- Improved window management functions
- Better terminal process handling
- Added follow mode improvements

#### Integration Improvements
- Centralized <Plug> mapping management
- Better integration discovery
- Health check support for all integrations

#### Command Processing
- Added mutually exclusive option validation
- Improved argument parsing
- Better completion support

## File Changes Summary

- **44 files changed**
- **3,600 insertions(+)**
- **2,264 deletions(-)**

### Major Files
- Module reorganization: 5 files split/renamed
- Test files: 15 files updated/added
- Documentation: 5 files updated
- Integration files: 4 files improved

## Testing

All tests pass successfully:
```
Test summary:
- Total tests: 135
- Passed: 135
- Failed: 0
```

### For Users
- Configuration key change: `agents` → `tools`
- Ftplugin files renamed: `aibo-agent-*` → `aibo-tool-*`

### For Developers
- Internal module paths changed
- Test helper functions renamed
- Some internal APIs reorganized

## Migration Guide

For users upgrading:
1. Update configuration: `agents = {}` → `tools = {}`
2. Update ftplugin references if customized
3. No other changes needed - functionality remains the same

## Documentation

All documentation has been updated:
- README.md
- CONTRIBUTING.md
- doc/aibo.txt
- doc/aibo-dev.txt
- REFACTORING_SUMMARY.md (new)

## Checklist

- [x] Tests pass (`just test`)
- [x] Linting passes
- [x] Documentation updated
- [x] Breaking changes documented
- [x] Migration guide provided

## Summary by CodeRabbit

- New Features
  - Revamped in-editor console and prompt windows with open/toggle/follow/send/submit, improved focus restore, and clearer notifications.
  - Smarter Aibo command completion and usage messages; AiboSend streamlined with range, prefix/suffix, replace, submit, and input options.

- Refactor
  - Terminology change: Agent → Tool. Configuration now uses tools, accessor renamed to get_tool_config, and mappings updated accordingly.

- Documentation
  - README and help docs updated to tool-centric language with revised examples and configuration guidance.

- Chores
  - Introduced a unified CI “Check” workflow; simplified lint/format/test pipeline and ignored local dependencies.

### CODE REVIEW

File: lua/aibo/integration/claude.lua [hunk 1]
```diff
@@ -284,49 +266,46 @@ end
...
+  end)
+  define("<Plug>(aibo-claude-shortcuts)", "Shortcuts (?)", function()
+    send("?")
+  end)
+  define("<Plug>(aibo-claude-bash-mode)", "Bash mode (!)", function()
+    send("!")
+  end)
+  define("<Plug>(aibo-claude-memorize)", "Memorize (#)", function()
+    send("#")
+  end)
```
[the remark this answers is not in the document]
Comment (lambdalisue) [reply lua/aibo/integration/claude.lua:308]: No. It's OK.
